core/src/abstractions.rs

```diff
@@ -32,6 +39,12 @@ impl MeteredSemaphore {
         res
     }
 
+    pub fn try_acquire_owned(&self) -> Result<OwnedSemaphorePermit, TryAcquireError> {
+        let res = self.sem.clone().try_acquire_owned();
+        (self.record_fn)(&self.metrics_ctx, self.sem.available_permits());
```

Comment:
Does it make sense to record if res is error?

Reply:
It's cheap. Branching just adds extra lines here.